File: day-1/day-1.py

def getMeThatNumber():
  counter = 0
  max = 0
  newlines = 0
  file = open("day-1-input.txt", "r")
  for line in file:
    if line != '\n':
      counter += int(line)
    else:
      if counter > max:
        max = counter
      counter = 0
  print('max', max)

getMeThatNumber()

File: day-1/day-1.js
// start by reading the input file
import * as fs from "fs";
import * as readline from "readline";

function getMeThatNumber() {

const fileStream = fs.createReadStream("day-1-input.txt");

let results = [];
let counter = 0;
let sum;
const rl = readline.createInterface({
  input: fileStream,
  crlfDelay: Infinity,
});

rl.on("line", (line) => {
  if (line) {
    counter += parseInt(line);
  } else {
    results.push(counter);
    counter = 0;
  }
});

rl.on("close", () => {
  results.push(counter);
  const last3 = results.sort((a, b) => a - b).slice(-3);
  // add the elements in the array together
  sum = last3.reduce((a, b) => a + b, 0);
  console.log(sum);
})


}
getMeThatNumber();


